tools/make.ps1

```diff
@@ -0,0 +1,130 @@
+$projectRoot  = Split-Path -Parent $PSScriptRoot
+$releasePage  = "https://github.com/KoffeinFlummi/armake/releases"
+$downloadPage = "https://github.com/KoffeinFlummi/armake/releases/download/v{0}/armake_v{0}.zip"
+$armake = "$projectRoot\tools\armake.exe"
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
+
+function Compare-VersionNewerThan {
+    param(
+        [Parameter(Mandatory=$True)]
+        $version1,
+
+        [Parameter(Mandatory=$True)]
+        $version2
+    )
+
+    $version1 = $version1.Split(".") | % {[int] $_}
+    $version2 = $version2.Split(".") | % {[int] $_}
+
+    $newer = $False
+    for ($i = 0; $i -lt $version1.Length; $i++) {
+        if ($version1[$i] -gt $version2[$i]) {
+            $newer = $True
+            break
+        }
+    }
+
+    $newer
+}
+
+
+function Get-InstalledArmakeVersion {
+    if (Test-Path $armake) {
+        $version = & $armake --version
+        $version = $version.Substring(1)
+    } else {
+        $version = "0.0.0"
+    }
+
+    $version
+}
+
+
+function Get-LatestArmakeVersion {
+    $client = New-Object Net.WebClient
+    $content = $client.DownloadString($releasePage)
+    $client.dispose()
+
+    $match = $content -match "<a href="".*?/releases/download/v(.*?)/.*?.zip"".*?>"
+    if (!$match) {
+        Write-Error "Failed to find valid armake download link."
+        $version = "0.0.0"
+    } else {
+        $version = $matches[1]
+    }
+
+    $version
+}
+
+
+function Update-Armake {
+    param(
+        [Parameter(Mandatory=$True)]
+        $url
+    )
+
+    New-Item "$PSScriptRoot\temp" -ItemType "directory" -Force | Out-Null
+
+    Write-Output "Downloading armake..."
+    $client = New-Object Net.WebClient
+    $client.DownloadFile($url, "$PSScriptRoot\temp\armake.zip")
+    $client.dispose()
+
+    Write-Output "Download complete, unpacking..."
+    Expand-Archive "$PSScriptRoot\temp\armake.zip" "$PSScriptRoot\temp\armake"
```

Comment:
Doesn't exist on windows 8.
Alternative: https://stackoverflow.com/a/27768628/6216977

Works for me.

Reply:
@BaerMitUmlaut ?